src/main/java/com/microsoft/lst_bench/client/JDBCConnection.java

```diff
@@ -74,31 +74,33 @@ private QueryResult execute(String sqlText, boolean ignoreResults) throws Client
         }
         // Log verbosely, if enabled.
         if (this.showWarnings && LOGGER.isWarnEnabled()) {
-          LOGGER.warn(createWarningString(s));
+          logWarnings(
+              s,
+              /* logPrefix= */ errorCount > 0 ? ("Retried query, error count: " + errorCount) : "");
         }
         // Return here if successful.
         return queryResult;
       } catch (Exception e) {
         queryResult = null;
         String lastErrorMsg =
-            "Query execution ("
-                + this.maxNumRetries
-                + " retries) unsuccessful; "
-                + "Warnings: "
-                + createWarningString(s)
-                + "stack trace: "
+            "Query execution attempt "
+                + (errorCount + 1)
+                + " unsuccessful, will retry "
+                + (this.maxNumRetries - errorCount)
+                + " more times; stack trace: "
                 + ExceptionUtils.getStackTrace(e);
-
-        // Log execution error and any pending warnings associated with this statement, useful for
-        // debugging.
         if (errorCount == this.maxNumRetries) {
+          // Log any pending warnings associated with this statement, useful for debugging.
+          if (LOGGER.isWarnEnabled()) {
```

Comment:
Consider moving this check inside the _logWarnings_ method and bailing out quickly if the warning is disabled. This way, the callers do not need to check it outside the method, e.g., second cond in L76.

Reply:
It seems this comment crossed your commit. If the method returns a `String`, then there is no need.